test/jdk/java/awt/print/PrinterJob/ExceptionFromPrintableIsIgnoredTest.java

```diff
@@ -0,0 +1,129 @@
+
+    public static void main(String[] args) {
+        if (args.length < 2) {
+            throw new RuntimeException("Two arguments are expected:"
+                    + " test thread type and test exception type.");
+        }
+
+        new ExceptionFromPrintableIsIgnoredTest(
+            TestThreadType.valueOf(args[0]),
+            TestExceptionType.valueOf(args[1]));
+    }
+
+    public ExceptionFromPrintableIsIgnoredTest(
+            final TestThreadType threadType,
+            final TestExceptionType exceptionType) {
+        System.out.println(String.format(
+                "Test started. threadType='%s', exceptionType='%s'",
+                threadType, exceptionType));
+
+        if (threadType == TestThreadType.MAIN) {
+            runTest(exceptionType);
+        } else if (threadType == TestThreadType.EDT) {
+            try {
+                SwingUtilities.invokeAndWait(new Runnable() {
+                    @Override
+                    public void run() {
+                        runTest(exceptionType);
+                    }
+                });
+            } catch (InterruptedException | InvocationTargetException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        System.out.println("Test passed.");
+    }
+
+    private void runTest(final TestExceptionType exceptionType) {
+        PrinterJob job = PrinterJob.getPrinterJob();
+        if (job.getPrintService() == null) {
+            throw new RuntimeException("No printers are available.");
+        }
+
+        job.setPrintable(new Printable() {
+            @Override
+            public int print(Graphics graphics, PageFormat pageFormat,
+                    int pageIndex) throws PrinterException {
+                if (pageIndex > 1) {
+                    return NO_SUCH_PAGE;
+                }
+                if (exceptionType == TestExceptionType.PE) {
+                    throw new PrinterException(
+                        "Exception from 'Printable.print'.");
+                } else if (exceptionType == TestExceptionType.RE) {
+                    throw new RuntimeException(
+                        "Exception from 'Printable.print'.");
+                }
+                return PAGE_EXISTS;
+            }
+        });
+
+        Throwable printEx = null;
+        try {
+            job.print();
+        } catch (Throwable t) {
+            printEx = t;
+        }
+
+        if (printEx != null) {
+            System.out.println("'PrinterJob.print' threw the exception:");
+            printEx.printStackTrace(System.out);
+
+            if (((printEx instanceof PrinterException) &&
+                    (exceptionType == TestExceptionType.PE)) ||
+                ((printEx instanceof RuntimeException) &&
+                    (exceptionType == TestExceptionType.RE))) {
+                return;// Test passed.
+            }
+            throw new RuntimeException("Unexpected exception was thrown.");
+        } else {
```

Comment:
When running on the EDT do you really want to do this ?

Reply:
Yes, I do really want to do this, when running on EDT, the whole test relies on this approach. When the test is run on EDT this is done via "SwingUtilities.invokeAndWait" method in the constructor of the test "ExceptionFromPrintableIsIgnoredTest" and in that constructor there is "try/catch" block. If some exception occurs on EDT and is not caught on EDT, then it will be wrapped by JDK in "InvocationTargetException" instance which will be thrown from "SwingUtilities.invokeAndWait" method and will be caught by that "try/catch" block in the constructor of the test. I checked that this code works before sending the code for review and today additionally.